homeassistant/util/logging.py

```diff
@@ -29,16 +29,21 @@ class HomeAssistantQueueListener(logging.handlers.QueueListener):
     LOG_COUNTS_RESET_INTERVAL = 300
     MAX_LOGS_COUNT = 200
 
+    EXCLUDED_LOG_COUNT_MODULES = [
+        "homeassistant.setup",
+        "homeassistant.components.automation",
+        "homeassistant.components.script",
+    ]
+
     _last_reset: float
     _log_counts: dict[str, int]
-    _warned_modules: set[str]
 
     def __init__(
         self, queue: SimpleQueue[logging.Handler], *handlers: logging.Handler
     ) -> None:
         """Initialize the handler."""
         super().__init__(queue, *handlers)
-        self._warned_modules = set()
+        self._module_log_count_skip_flags: dict[str, bool] = {__name__: True}
```

Comment:
Why do we add ourselves? Would it be clearer to add `homeassistant.util.logging` to the exclude-list?

Reply:
moved to the exclude list.
if we don't add ourselves we could get a "logging too frequently" for this module if many others were logging too often.
this was more of an issue with the initial implementation, before having the 1-warning-per-module limit, so I am OK with removing it now from the exclusions if you prefer.